packages/react/src/auto/polaris/PolarisAutoTable.tsx

```diff
@@ -124,7 +124,18 @@ export const PolarisAutoTable = <
         {rows &&
           columns &&
           rows.map((row, index) => (
-            <IndexTable.Row key={row.id as string} id={row.id as string} position={index}>
+            <IndexTable.Row
+              key={row.id as string}
+              id={row.id as string}
+              position={index}
+              onClick={
+                props.onClick
+                  ? () => {
```

Comment:
This has the potential to create unnecessary re-renders since it will allocate a new anonymous function each time. You could pull it up to a useCallback() variable maybe?